noir/noir-repo/acvm-repo/acvm/src/pwg/mod.rs

```diff
@@ -556,6 +566,21 @@ fn any_witness_from_expression(expr: &Expression) -> Option<Witness> {
     }
 }
 
+/// Returns `true` if the predicate is zero
+/// A predicate is used to indicate whether we should skip a certain operation.
+/// If we have a zero predicate it means the operation should be skipped.
+pub fn is_predicate_false(
+    witness: &WitnessMap,
+    predicate: &Option<Expression>,
+) -> Result<bool, OpcodeResolutionError> {
+    // If the predicate is `None`, then we simply return the value 1
+    let pred_value = match predicate {
+        Some(pred) => get_value(pred, witness)?,
+        None => FieldElement::one(),
+    };
+    Ok(pred_value.is_zero())
```

Comment:
```suggestion
    let pred_value = match predicate {
        Some(pred) => get_value(pred, witness).map(|pred_value| pred_value.is_zero()),
        // If the predicate is `None`, then we treat it as an unconditional `true`
        None => Ok(false),
    }
```

We can skip constructing a `FieldElement` here.